lib/src/theme/color_extensions.dart

```diff
@@ -0,0 +1,283 @@
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'color_swatch.dart';
+import 'constants.dart';
+import 'contrast.dart';
+
+/// Extensions on [Color] to brighten, lighten, darken and blend colors and
+/// can get a shade for gradients.
+///
+/// Some of the extensions are rewrites of TinyColor's functions
+/// https://pub.dev/packages/tinycolor. The TinyColor algorithms have also
```

Comment:
As we have modified this code, we should include it's license in the repo. Should we have a third party license file like zds?

Reply:
That's a good idea. I'll pull it from the Zds repo.

Reply:
I've included the LICENSE-3RD-PARTY file and have documented the use of tinycolor and IBMPlexSans within it.